src/server/stream_family.cc

```diff
@@ -2817,7 +2817,27 @@ void XReadBlock(ReadOpts opts, ConnectionContext* cntx) {
   auto tp = (opts.timeout) ? chrono::steady_clock::now() + chrono::milliseconds(opts.timeout)
                            : Transaction::time_point::max();
 
-  if (auto status = cntx->transaction->WaitOnWatch(tp, std::move(wcb)); status != OpStatus::OK)
+  const auto& keyChecker = [&opts](EngineShard* owner, const DbContext& context, Transaction* tx,
```

Comment:
curious why `const auto&` and not `const auto`? seems like unneeded complexity, since you introduce const reference to a temporary expression.

Reply:
there are no differences from my point of view, but I will update ir